ipc/api/src/checkpoint.rs

```diff
@@ -53,22 +57,27 @@ impl Display for QuorumReachedEvent {
 pub struct BottomUpCheckpointBundle {
     pub checkpoint: BottomUpCheckpoint,
     /// The list of signatures that have signed the checkpoint hash
+    #[serde(deserialize_with = "deserialize_vec_bytes_from_vec_hex")]
+    #[serde(serialize_with = "serialize_vec_bytes_to_vec_hex")]
     pub signatures: Vec<Signature>,
```

Comment:
`serde_as` can actually be used in combination with containers. For example further down you have this:
```rust
    #[serde_as(as = "HumanReadable")]
    pub block_hash: Vec<u8>,
```
But here we have a `Vec<Vec<u8>>`. In this situations it can be used like [here](https://github.com/consensus-shipyard/ipc/blob/a33cfee763cafaafdd00ad38a1367ee262e75fa4/fendermint/app/settings/src/resolver.rs#L58-L59): 

```rust
    #[serde_as(as = "Vec<HumanReadable>")]
    pub signatures: Vec<Signature>,
```

It also works with `Option`, etc.

Reply:
It would be nicer because then in normal encoding it would still use the potentially more space efficient byte encoding (although at this point I'm not actually sure if for that we need to involve the `strict_bytes` module. But at the very least it would be a number, rather than 2 characters per byte.

Reply:
By contrast I think this might be backwards incompatible: if it met an existing record encoded as IPLD bytes it would look for a `String`.

Reply:
Nice tip, I tried and it works. I have updated accordingly to use:
```
#[serde_as(as = "Vec<HumanReadable>")]
pub signatures: Vec<Signature>,
```